### PR TITLE
dist: improve scripts and fix build container permissions

### PR DESCRIPTION
**Pull request by steveej**
/cc @riuvshin

### CODE REVIEW

File: dist/build/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM clux/muslrust:1.30.0-stable
```
Comment (riuvshin) [inline dist/build/Dockerfile:1]: in previous PR you wanted newer image maybe it worth to use newer here?

Comment (steveej) [reply dist/build/Dockerfile:1]: The version bump is in #25. I wanted to split it apart but it looks like the upstream image changes make it difficult.

**Comment by steveej** [comment]
[test]

File: dist/commons.sh [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+function docker_cargo_stop_all() {
+    containers=$(docker ps --quiet --filter "label=${DOCKER_CARGO_LABEL}")
+    if test "${containers}"; then
+        docker stop ${containers} 2>&1 >/dev/null
+    fi
+}
+
+function ensure_build_container() {
+    docker build -t "${IMAGE_BUILD}" dist/build/
+}
```
Comment (riuvshin) [inline dist/commons.sh]: missing new line

**Pull request merged**